### ISSUE TITLE
Error with IndexHierarchy `via_fill_value` item access

### ISSUE THREAD

**Issue by gettodaze** [static-frame/static-frame#1004]
**Description**
`static_frame.core.node_fill_value.InterfaceFillValue.__getitem__` fails where `sf.Frame.__getitem__` succeeds.

**Example**

```python
label = (1, 'a')
labels = [label]
f = sf.Frame.from_element(0, index=['x'], columns=sf.IndexHierarchy.from_labels(labels))
print("f")
print(f)
print("f[labels]")
print(f[labels]) # works
print("f.via_fill_value(-1)[label]")
print(f.via_fill_value(-1)[label]) # works
print("f.via_fill_value(-1)[labels]")
print(f.via_fill_value(-1)[labels]) # errors
```
```
f
<Frame>
<IndexHierarchy> 1       <int64>
                 a       <<U1>
<Index>
x                0
<<U1>            <int64>
f[labels]
<Frame>
<IndexHierarchy> 1       <int64>
                 a       <<U1>
<Index>
x                0
<<U1>            <int64>
f.via_fill_value(-1)[label]
<Series: (1, 'a')>
<Index>
x                  0
<<U1>              <int64>
f.via_fill_value(-1)[labels]
---------------------------------------------------------------------------
ErrorInitIndex                            Traceback (most recent call last)
Cell In[17], line 11
      9 print(f.via_fill_value(-1)[label]) # works
     10 print("f.via_fill_value(-1)[labels]")
---> 11 print(f.via_fill_value(-1)[labels]) # errors

File ~/.env311/lib/python3.11/site-packages/static_frame/core/node_fill_value.py:234, in InterfaceFillValue.__getitem__(self, key)
    232 if self._container._NDIM == 1:
    233     return self._extract_loc1d(key)
--> 234 return self._extract_loc2d(NULL_SLICE, key)

File ~/.env311/lib/python3.11/site-packages/static_frame/core/node_fill_value.py:170, in InterfaceFillValue._extract_loc2d(self, row_key, column_key)
    167 if row_is_multiple and column_is_multiple:
    168     # cannot reindex if loc keys are elements
    169     index = index_from_index(row_key, container.index) if not row_is_null_slice else None
--> 170     columns = index_from_index(column_key, container.columns) if not column_is_null_slice else None
    172     return container.reindex(
    173             index=index,
    174             columns=columns,
   (...)
    177             own_columns=columns is not None,
    178             )
    179 elif not row_is_multiple and not column_is_multiple: # selecting an element

File ~/.env311/lib/python3.11/site-packages/static_frame/core/container_util.py:517, in index_from_index(value, index)
    515 ctr = partial(index.__class__, name=index.name)
    516 setattr(ctr, STATIC_ATTR, getattr(index, STATIC_ATTR))
--> 517 return index_from_optional_constructor(value, default_constructor=ctr)

File ~/.env311/lib/python3.11/site-packages/static_frame/core/container_util.py:509, in index_from_optional_constructor(value, default_constructor, explicit_constructor)
    506         return value._MUTABLE_CONSTRUCTOR(value)
    508 # cannot always determine static status from constructors; fallback on using default constructor
--> 509 return default_constructor(value)

File ~/.env311/lib/python3.11/site-packages/static_frame/core/index_hierarchy.py:1028, in IndexHierarchy.__init__(self, indices, indexers, name, blocks, own_blocks)
   1025     raise ErrorInitIndex('indexers must be a read-only numpy array.')
   1027 if not all(isinstance(index, Index) for index in indices):
-> 1028     raise ErrorInitIndex("indices must all Index's!")
   1030 if len(indices) <= 1:
   1031     raise ErrorInitIndex('Index Hierarchies must have at least two levels!')

ErrorInitIndex: indices must all Index's!
```

**Platform**

Run the following function (static-frame >= 0.8.1) and provide the results to define your platform and environment:

```python
>>> import static_frame as sf
>>> sf.Platform.display()
<Series: platform>
<Index>
platform           Linux-5.15.0-125-generic-x86_64-with-glibc2.31
sys.version        3.11.4 (main, Jun 24 2024, 10:33:32) [GCC 9.4.0]
static-frame       2.14.0
numpy              1.24.3
pandas             1.5.3
xlsxwriter         3.1.0
openpyxl           3.0.9
xarray             <ModuleNotFoundError>
tables             3.8.0
pyarrow            12.0.1
msgpack            None
msgpack_numpy      None
<<U13>             <object>
```

**Comment by flexatone** [comment]
Thanks, @gettodaze , for isolating this issue. I will investigate a fix.

**Comment by flexatone** [comment]
The fix will be part of the 2.15.1 release.
